### PR TITLE
feat(gateway): tell the agent Lore manages the context window

### PR DESCRIPTION
**Pull request by BYK**
## Problem

Suggested by Onur: coding agents over-hedge on large tasks — "I can't do this, we only have limited context", "this session is getting long, should we stop?" — forcing the user to repeat "we use Lore / auto-caching / don't worry about context" every session. Lore should proactively tell the agent the real situation so it knows the true scope it can take on.

## Fix

Prepend a short, **static** capability note (`LORE_CONTEXT_CAPABILITY_NOTE`) to the frozen `system[1]` block:

> Lore actively manages and compresses this session's context and preserves older turns as recall-able summaries, so your effective context is far larger than it looks. Don't hedge or stop over context limits; take on large, multi-step tasks directly.

### Why static / why system[1]

A per-turn "Context health" note whose wording varied by compression layer used to live in `system[2]` and busted the conversation cache on every layer oscillation (issue #741). This note is safe precisely because it **never changes**: no token counts, no layer names. It's frozen with the `system[1]` baseline (1h cache) and present from turn 1, so it's a stable cache read every turn. Because it's always present, `system[1]` no longer has an "empty baseline" edge case (the mid-session-appearance bust the freeze guarded against can't occur).

## Tests

- Static-content guard: the note contains none of the #741-forbidden strings (`aggressively compressed`, `emergency compressed`, `[Context health:`) and no digits (a token count/layer number would make it vary per turn).
- Harness test: the note lands in `system[1]` on turn 1 with **no** knowledge/entities seeded (proves always-present). **Mutation-verified** — removing the note from the assembly fails this test.
- `cache-stability.e2e` still asserts byte-identical system blocks across forced layer escalation (1→2→3); `context-health-note` guardrail still green. typecheck + oxlint + oxfmt clean.

Closes the loop on Onur's suggestion. (His related "say it 1–2× → auto-LTM" ask is the pattern-extract/curator path; this note makes the behavior work without depending on that capture.)

### CODE REVIEW

File: packages/gateway/src/pipeline.ts [hunk 1]
```diff
@@ -6887,22 +6907,27 @@ async function handleConversationTurn(
...
+        // with this baseline, system[1] is now always present and byte-stable —
+        // which also means the "empty baseline" case below can no longer occur.
+        const formatted = [
+          LORE_CONTEXT_CAPABILITY_NOTE,
+          prefText,
+          entitiesText,
+          knowledgeTocText,
+        ]
           .filter(Boolean)
           .join("\n\n");
```
[the remark this answers is not in the document]
Comment (BYK) [reply packages/gateway/src/pipeline.ts:6922]: Accurate observation, but this is intended and the cache-safe choice — not a bug.

system[1] is frozen per session by design (v45): it is never recomputed from the live state mid-session, precisely so the 1h-cached prefix stays byte-identical. A pre-existing session restored from the DB keeps its exact persisted baseline and gets a cache *read* every turn. It picks up the note the next time it starts a fresh session — identical to how every other system[1] change propagates (e.g. a mid-session curated preference only appears next session).

The suggested fix (prepend the note into the restored baseline) would change those sessions' system[1] bytes on their first post-deploy turn, forcing a one-time cache bust across every active session for a soft capability hint. That directly contradicts the "never bust / never recompute system[1] mid-session" invariant this block is built around, so I'm intentionally not doing it. New sessions get the note from turn 1; old sessions self-heal as they cycle, with zero cache disruption.
